Gems/LyShine/Code/Editor/EditorWindow.cpp

```diff
@@ -697,15 +698,27 @@ bool EditorWindow::SaveCanvasToXml(UiCanvasMetadata& canvasMetadata, bool forceA
         else if (recentFiles.size() > 0)
         {
             dir = Path::GetPath(recentFiles.front());
-            dir.append(canvasMetadata.m_canvasDisplayName.c_str());
         }
         // Else go to the default canvas directory
         else
         {
             dir = FileHelpers::GetAbsoluteDir(UICANVASEDITOR_CANVAS_DIRECTORY);
-            dir.append(canvasMetadata.m_canvasDisplayName.c_str());
         }
 
+        // Make sure the directory exists. If not, walk up the directory path until we find one that does
+        // so that we will have a consistent 'starting folder' in the 'AzQtComponents::FileDialog::GetSaveFileName' call
+        // across different platforms. 
+        AZ::IO::BasicPath<AZStd::string> dirPath(dir.toUtf8().constData());
```

Comment:
Just use `AZ::IO::FixedMaxPath` or `AZ::IO::Path`

Reply:
I'll remove the use of QDir and update this section. But the final check is not needed, because if the root does not exist (i.e. if its on a drive that is not on the windows machine), the default behavior of the save dialog should handle that in its own way.